Library/Homebrew/extend/os/mac/diagnostic.rb

```diff
@@ -129,11 +129,10 @@ def check_for_unsupported_macos
         end
 
         def check_for_opencore
-          has_opencore =
-            File.exist?("/Library/PrivilegedHelperTools/com.dortania.opencore-legacy-patcher.privileged-helper")
-          has_opencore ||= File.exist?("/Library/Application Support/Dortania/OpenCore-Patcher.app")
-          has_opencore ||= MacOS.pkgutil_info("com.dortania.opencore-legacy-patcher").present?
-          return unless has_opencore
+          return if ::Hardware::CPU.physical_cpu_arm64?
+
+          loaded_kexts = Utils.safe_popen_read("/usr/bin/kmutil", "showloaded")
+          return if loaded_kexts.exclude?("FakeSMC") && loaded_kexts.exclude?("VirtualSMC")
```

Comment:
Unfortunately this check also flags VMs, so doesn't really work 😢

https://github.com/Homebrew/brew/actions/runs/11816737888/job/32920685713?pr=18764#step:4:14